external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -447,6 +447,8 @@ void outlinePartitionOps(Operation *anchorOp, ArrayRef<Operation *> trailingOps,
 void TosaPartitionPass::specialCaseForTranspose(Operation *op,
                                                 SetVector<Operation *> &ops) {
   auto *operand = op->getOpOperand(1).get().getDefiningOp();
+  if (ops.contains(operand)) // If already present, move it for new use.
+    ops.remove(operand);
```

Comment:
The problem situation was two TransposeOps that used the same constant.  Tracing backward from the anchor, we'd save one transpose, then its constant (because it was tracing depth-first), and then the other transpose and its constant.  But the constant was already saved, so we wouldn't save it again, and it would end up between the two transposes instead of ahead of both.

By removing it, the usual insertion always places the constant ahead of all its uses.